src/main/java/com/elastic/support/BaseConfig.java

```diff
@@ -1,35 +1,57 @@
 package com.elastic.support;
 
+import org.apache.commons.lang3.StringUtils;
+
 import java.util.Map;
 
 public class BaseConfig {
 
-    protected Map configuration;
-    protected Map<String, String> githubSettings;
-    protected Map<String, Integer> restConfig;
+    public String diagnosticVersion;
+    public int connectionTimeout;
+    public int connectionRequestTimeout;
+    public int socketTimeout;
+    public int maxTotalConn;
+    public int maxConnPerRoute;
 
-    public BaseConfig() {
-    }
+    public String diagReleaseHost = "api.github.com";
```

Comment:
I actually plan to either remove these from the configuration or comment them. The only time they'd need to be in the config is if something changed prior to a release and you needed a way to modify the config as a workaround. But you do then get that "get out of jail free card". Had intended to make a final pass through and possibly make some additional changes to the config in a future change.

And on that note, if you make it final it seems that stops you from setting it via config. And in this particular case, it doesn't hurt I guess, but what's going to change it?  What set of circumstances would cause somebody to do that? Usually if I can't easily think of one in 15 minutes or so I don't worry about it.